sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -191,6 +214,16 @@ class MapboxMap internal constructor(
     styleExtension: StyleContract.StyleExtension
   ) = loadStyle(styleExtension, null, null)
 
+  /**
+   * Set style transition options that will be applied to all style loads happening after.
+   *
+   * In order to obtain current transition options please use [Style.getStyleTransition]
+   * when style will be loaded.
+   */
+  fun setStyleTransition(transitionOptions: TransitionOptions) {
+    styleTransitionOptions = transitionOptions
+  }
```

Comment:
what if the style is loaded and then the `setStyleTransition` is applied? The setStyleTransition will never go through to gl-native until the next style is loaded..

Continuing my previous question in https://github.com/mapbox/mapbox-maps-android/pull/1174/files#r812699285

> I mean does the setStyleTransition change the default transition for all the layer properties when the property is changed after the style is fully loaded?

If setting the `StyleTransition` changes the default transition for all the transitional style properties, we should apply the `setStyleTransition` to gl-native immediately if the Style is loaded.

Otherwise this function will introduce bug when `MapboxMap.setStyleTransition` is called after the style is loaded, and some of the transitionable style properties is changed, while the default transition is not applied.

Reply:
thanks, @pengdev, good point